### PR TITLE
Fix SQL error introduced by da10f001

### LINKED ISSUE #30833: PS 8.0.1. class/Customer.php function getSimpleAddressSql() generates incorrect SQL
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

In 8.0.1 was added ORDER BY expression to class/Customer.php function getSimpleAddressSql().
But it was not considered next optional adding expression for filtering by id_address.
Condition "AND a.`id_address` = xxx" ignored because of incorrect SQL statement "ORDER BY a.`alias` AND a.`id_address` = xxx":

```
SELECT DISTINCT
                      a.`id_address` AS `id`,
                      a.`alias`,
                      a.`firstname`,
                      a.`lastname`,
                      a.`company`,
                      a.`address1`,
                      a.`address2`,
                      a.`postcode`,
                      a.`city`,
                      a.`id_state`,
                      s.name AS state,
                      s.`iso_code` AS state_iso,
                      a.`id_country`,
                      cl.`name` AS country,
                      co.`iso_code` AS country_iso,
                      a.`other`,
                      a.`phone`,
                      a.`phone_mobile`,
                      a.`vat_number`,
                      a.`dni`
                    FROM `ps_address` a
                    LEFT JOIN `ps_country` co ON (a.`id_country` = co.`id_country`)
                    LEFT JOIN `ps_country_lang` cl ON (co.`id_country` = cl.`id_country`)
                    LEFT JOIN `ps_state` s ON (s.`id_state` = a.`id_state`)
                     INNER JOIN ps_country_shop country_shop
        ON (country_shop.id_country = co.id_country AND country_shop.id_shop = 1)
                    WHERE
                        `id_lang` = 1
                        AND `id_customer` = 2
                        AND a.`deleted` = 0
                        AND a.`active` = 1
                    ORDER BY a.`alias` AND a.`id_address` = 16663
```
And as result will be selected all the customer addresses, but not the one filtered by id_address. 

### Expected behavior

function getSimpleAddressSql() should return one selected address filtered by id_address. 
Correct generated SQL code should be the following:
```
SELECT DISTINCT
                      a.`id_address` AS `id`,
                      a.`alias`,
                      a.`firstname`,
                      a.`lastname`,
                      a.`company`,
                      a.`address1`,
                      a.`address2`,
                      a.`postcode`,
                      a.`city`,
                      a.`id_state`,
                      s.name AS state,
                      s.`iso_code` AS state_iso,
                      a.`id_country`,
                      cl.`name` AS country,
                      co.`iso_code` AS country_iso,
                      a.`other`,
                      a.`phone`,
                      a.`phone_mobile`,
                      a.`vat_number`,
                      a.`dni`
                    FROM `ps_address` a
                    LEFT JOIN `ps_country` co ON (a.`id_country` = co.`id_country`)
                    LEFT JOIN `ps_country_lang` cl ON (co.`id_country` = cl.`id_country`)
                    LEFT JOIN `ps_state` s ON (s.`id_state` = a.`id_state`)
                     INNER JOIN ps_country_shop country_shop
        ON (country_shop.id_country = co.id_country AND country_shop.id_shop = 1)
                    WHERE
                        `id_lang` = 1
                        AND `id_customer` = 2
                        AND a.`deleted` = 0
                        AND a.`active` = 1
                        AND a.`id_address` = 16663
                    ORDER BY a.`alias`
```
### Steps to reproduce

php call $customer->getSimpleAddress($id_address);

### PrestaShop version(s) where the bug happened

8.0.1

### PHP version(s) where the bug happened

7.4.20

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by lmeyer1**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.0.x
| Description?      | Fixes SQL error introduced with da10f001
| Type?             | bug fix
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #30833
| Related PRs       | None
| How to test?      | Please indicate how to best verify that this PR is correct.
| Possible impacts? | Please indicate what parts of the software we need to check to make sure everything is alright.

### CODE REVIEW

**Review by FabienPapet — APPROVED** [review]

**Review by nicosomb — APPROVED** [review]

**Review by PululuK — APPROVED** [review]

**Review by PrestaEdit — APPROVED** [review]

**Comment by florine2623** [comment]
@lmeyer1 could you indicate how to test this PR for the QA team please ?

File: classes/Customer.php [hunk 1]
```diff
@@ -785,13 +785,14 @@ public function getSimpleAddressSql($idAddress = null, $idLang = null)
...
-                    ORDER BY a.`alias`';
+                        AND a.`active` = 1';
 
         if (null !== $idAddress) {
             $sql .= ' AND a.`id_address` = ' . (int) $idAddress;
         }
 
+        $sql .= ' ORDER BY a.`alias`';
+
         return $sql;
```
Comment (lmeyer1) [inline classes/Customer.php:796]: To test insert here `echo $sql;`, just before the return statement

**Comment by lmeyer1** [comment]
Here are the steps for QA:
1. Prove that the queries in #30833 are (1) wrong in the section 'Bug' and right in the section 'Expected'
    - Select a valid id_customer with id_address of an associated address (id_address != 16663)
    - Copy both the queries into phpmyadmin, replace the id_customer with the one you selected, let the id_address (that does not belong to the customer.
    - Expected: empty result (since the address 16663 does not belong to the selected customer. Result: all adresses of customer for first query: wrong, empty result for second query: correct
    - Copy both the queries into phpmyadmin, replace the id_adress and the id_customer with the one you selected.
    - Expected: the address with id `id_adress`. Result: all adresses of customer for first query: wrong, correct address for second query: correct
2. Prove that the queries are those produced with the code prior to PR and after PR
    - Insert `echo $sql;` here: https://github.com/PrestaShop/PrestaShop/pull/30834#discussion_r1069446611
    - Insert `if ($this->context->customer->id) {var_dump($this->context->customer->getSimpleAddress(-1)); die();}` for instance here https://github.com/PrestaShop/PrestaShop/blob/develop/classes/controller/FrontController.php#L265. `-1` is an inexistent id_address, we therefore should get an empty array as result
    - Navigate to any page, log in if your not already logged.
    - You will see an array of all addresses of the currently logged in user, instead of an empty array. After the PR, the result will be an empty array: correct
    - Replace the id_address `-1` with the id of an address belonging to the logged in user
    - You will see an array of all addresses of the currently logged in user, instead of an empty array. After the PR, the result will be an array containing only the expected address: correct

**Review by davidglezz — APPROVED** [review]

**Comment by Hlavtox** [comment]
Works perfectly @lmeyer1 ;-)
Before:
BEFORE

After:
after

**Comment by lmeyer1** [comment]
@Hlavtox Thank you very much !

**Pull request merged**